users/user_permission.go

```diff
@@ -2,115 +2,101 @@ package users
 
 import (
 	"context"
-	"fmt"
+	"encoding/json"
 	"net/http"
 
 	"github.com/SKF/go-rest-utility/client"
-	"github.com/go-http-utils/headers"
+	"github.com/SKF/go-utility/v2/array"
 	"github.com/pkg/errors"
 )
 
+type updateRoleFunc func(roles []string, roleToUpdate string) []string
+
 func AddUserRole(identityToken, stage, userID, role string) error {
 	return AddUserRoleWithContext(context.Background(), identityToken, stage, userID, role)
 }
 
 func AddUserRoleWithContext(ctx context.Context, identityToken, stage, userID, role string) (err error) {
-	user, err := getUser(ctx, identityToken, stage, userID)
-	if err != nil {
-		return
-	}
-
-	user.UserRoles = append(user.UserRoles, role)
-	return updateUser(ctx, identityToken, stage, user)
+	return updateRoleToAllUsersNodes(ctx, identityToken, stage, userID, role, addRole)
 }
 
 func RemoveUserRole(identityToken, stage, userID, role string) error {
-	return RemoveUserRoleWithContext(context.Background(), identityToken, stage, userID, role)
+	return updateRoleToAllUsersNodes(context.Background(), identityToken, stage, userID, role, removeRole)
 }
 
-func RemoveUserRoleWithContext(ctx context.Context, identityToken, stage, userID, roleToBeRemoved string) (err error) {
```

Comment:
Is it safe to remove this function without a major version bump?

Reply:
Good catch, I reintroduce it.